Src/Extern/HDF5/AMReX_WriteBinaryParticleDataHDF5.H

```diff
@@ -0,0 +1,459 @@
+#ifndef AMREX_WRITE_BINARY_PARTICLE_DATA_HDF5_H
+#define AMREX_WRITE_BINARY_PARTICLE_DATA_HDF5_H
+
+static int CreateWriteHDF5Attr(hid_t loc, const char *name, hsize_t n, void *data, hid_t dtype)
+{
+    herr_t ret;
+    hid_t attr, attr_space;
+    hsize_t dims = n;
+
+    attr_space = H5Screate_simple(1, &dims, NULL);
+
+    attr = H5Acreate(loc, name, dtype, attr_space, H5P_DEFAULT, H5P_DEFAULT);
+    if (attr < 0) {
+        amrex::Print() << " Error with H5Acreate [" << name << "]\n";
+        return -1;
+    }
+
+    ret  = H5Awrite(attr, dtype, (void*)data);
+    if (ret < 0) {
+        amrex::Print() << " Error with H5Awrite [" << name << "]\n";
+        return -1;
+    }
+    H5Sclose(attr_space);
+    H5Aclose(attr);
+    return 1;
+}
+
+static int CreateWriteHDF5AttrString(hid_t loc, const char *name, const char* str)
+{
+    hid_t attr, atype, space;
+    herr_t ret;
+
+    AMREX_ASSERT(name);
+    AMREX_ASSERT(str);
+
+    space = H5Screate(H5S_SCALAR);
+    atype = H5Tcopy(H5T_C_S1);
+    H5Tset_size(atype, strlen(str)+1);
+    H5Tset_strpad(atype,H5T_STR_NULLTERM);
+    attr = H5Acreate(loc, name, atype, space, H5P_DEFAULT, H5P_DEFAULT);
+    if (attr < 0) {
+        amrex::Print() << " Error with H5Acreate [" << name << "]\n";
+        return -1;
+    }
+
+    ret = H5Awrite(attr, atype, str);
+    if (ret < 0) {
+        amrex::Print() << " Error with H5Awrite [" << name << "]\n";
+        return -1;
+    }
+
+    H5Tclose(atype);
+    H5Sclose(space);
+    H5Aclose(attr);
+
+    return 1;
+}
+
+static int ReadHDF5Attr(hid_t loc, const char *name, void *data, hid_t dtype)
+{
+    herr_t ret;
+    hid_t attr;
+
+    attr = H5Aopen(loc, name, H5P_DEFAULT);
+    if (attr < 0) {
+        amrex::Print() << " Error with H5Aopen [" << name << "]\n";
+        return -1;
+    }
+
+    ret  = H5Aread(attr, dtype, data);
+    if (ret < 0) {
+        amrex::Print() << " Error with H5Aread [" << name << "]\n";
+        return -1;
+    }
+    H5Aclose(attr);
+    return 1;
+}
+
+#ifdef BL_USE_MPI
```

Comment:
We should add `#include <AMReX_Config.H>` at the beginning of this header, so that `BL_USE_MPI` is always defined.

Reply:
This file is only included in Src/Particle/AMReX_WriteBinaryParticleData.H which has #include <AMReX_Config.H>